jina/serve/runtimes/head/__init__.py

```diff
@@ -262,28 +280,39 @@ async def process_control(self, request: ControlRequest, *args) -> ControlReques
 
     async def endpoint_discovery(self, empty, context) -> jina_pb2.EndpointsProto:
         """
-        USes the connection pool to send a discover endpoint call to the workers
+        Uses the connection pool to send a discover endpoint call to the workers
 
         :param empty: The service expects an empty protobuf message
         :param context: grpc context
         :returns: the response request
         """
         response = jina_pb2.EndpointsProto()
-        if self.uses_before_address:
-            uses_before_response, _ = await self.connection_pool.send_discover_endpoint(
-                deployment='uses_before', head=False
+        try:
```

Comment:
where is this try closed?

Reply:
In line 312, it catches our `InternalNetworkError`